{{cookiecutter.project_name}}/pyproject.toml

```diff
@@ -1,30 +1,49 @@
-[tool.poetry]
+[build-system]
+requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]
+build-backend = "poetry_dynamic_versioning.backend"
+
+[project]
 name = "{{cookiecutter.__project_slug}}"
-version = "0.1.0"
 description = "{{cookiecutter.project_description}}"
 authors = ["{{cookiecutter.__author}}"]
 license = "{{cookiecutter.license}}"
 readme = "README.md"
 include = ["README.md", "src/{{cookiecutter.__project_slug}}/schema", "project"]
 
+requires-python = ">=3.9"
+
+dynamic = ["version"]
+
+dependencies = [
+  "linkml-runtime >=1.8.0",
+]
+
+[project.optional-dependencies]
+docs = [
+  "mkdocs-material >=8.2.8",
+  "mkdocs-mermaid2-plugin >=1.1.1",
+  "schemasheets >= 0.1.14",
+]
+dev = [
+  # "pid4cat_model[docs]",  # no support for recursive optional dependencies: poetry#3369
+  "mkdocs-material >=8.2.8",
+  "mkdocs-mermaid2-plugin >=1.1.1",
+  "schemasheets >= 0.1.14",
+  # end repeated docs extra
+  "linkml >= 1.8.5",
+]
+[tool.poetry]
+requires-poetry = ">=2.0"
+version = "0.0.0"
+
 [tool.poetry.dependencies]
-python = "^3.9"
-linkml-runtime = "^1.1.24"
+python = ">=3.9,<4.0"  # poetry needs an upper bound for locking
```

Comment:
My instinct is to make this a tighter upper-bound, i.e. <3.14, possibly <3.15. Feature development releases feel like they could be breaking and I wouldn't want to get caught up in that.

Reply:
I may also not understand the Python release cycle as well as I think and this is probably just fine.

Reply:
It was not fixed before and I am more for less strict 😉 . We can change in the future if needed.